### PR TITLE
UI: remove vuex module for notifaction and refactor component structure

### PR DESCRIPTION
**Pull request by georg-schwarz**
Please also review by starting the UI with backend and testing manually

Changes:
* remove unnecessary vuex module, use rest interface instead
* change frontend notification config model to have parameters instead of plain interface
* extract NotificationForm component and route to NotificationCreate and NotificationEdit components (no if-else clutter anymore)
* Refactor slack / webhook firebase config handling into corresponding form components only dealing with the parameters
* linter fixes
* better use of watchers

### CODE REVIEW

File: ui/src/notification/NotificationForm.vue [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    <v-row
+      cols="10"
+      sm="6"
+      md="4"
+    >
+      <webhook-notification-form
+        v-if="notification.type === 'webhook'"
+        v-model="notification.parameters"
+        style="flex: 1 1 auto"
+        @validityChanged="isValid = $event"
```
Comment (georg-schwarz) [inline ui/src/notification/NotificationForm.vue]: not sure if that really works. Better separate and join form validation of this component with parameters via &&

File: ui/src/notification/FirebaseNotificationForm.vue [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  private initialValidityCheck (): void {
+    this.emitIsValid()
+  }
+
+  @Emit('value')
+  emitValue (): FirebaseNotificationParameters {
+    return this.parameters
+  }
+
+  @Emit('changeValidity')
```
Comment (andreas-bauer) [inline ui/src/notification/FirebaseNotificationForm.vue:55]: I changed the event name to `changeValidity` to be more consistent with other event names, such as `click` or `change`.

File: ui/src/notification/FirebaseNotificationForm.vue [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  emitIsValid (): boolean {
+    return this.isValid
+  }
+
+  @Watch('parameters', { deep: true })
+  onChangeFormModel (): void {
```
Comment (andreas-bauer) [inline ui/src/notification/FirebaseNotificationForm.vue:61]: Also changed the name of the handler method to `onChangeFormModel` to be more consistent.

File: ui/src/notification/FirebaseNotificationForm.vue [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+@Component({ })
+export default class FirebaseNotificationForm extends Vue {
+  @PropSync('value')
+  private parameters!: FirebaseNotificationParameters
+
+  private isValid = false
+
+  private mounted (): void {
+    this.initialValidityCheck()
```
Comment (andreas-bauer) [inline ui/src/notification/FirebaseNotificationForm.vue:43]: I replaced the comment with a method that does communicate the purpose of the `this.emitIsValid()` call without the need of a comment.

**Review by andreas-bauer — DISMISSED** [review]

**Review by andreas-bauer — APPROVED** [review]

**Pull request merged**